manager/src/main/java/io/gingersnapproject/k8s/CacheRuleInformer.java

```diff
@@ -0,0 +1,410 @@
+package io.gingersnapproject.k8s;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.event.Event;
+import javax.enterprise.event.Observes;
+import javax.enterprise.inject.Instance;
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.google.protobuf.util.JsonFormat;
+
+import io.fabric8.kubernetes.api.model.ConfigMap;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.dsl.Resource;
+import io.fabric8.kubernetes.client.informers.ResourceEventHandler;
+import io.fabric8.kubernetes.client.informers.SharedIndexInformer;
+import io.gingersnapproject.configuration.Connector;
+import io.gingersnapproject.configuration.EagerRule;
+import io.gingersnapproject.configuration.LazyRule;
+import io.gingersnapproject.k8s.configuration.KubernetesConfiguration;
+import io.gingersnapproject.proto.api.config.v1alpha1.EagerCacheRuleSpec;
+import io.gingersnapproject.proto.api.config.v1alpha1.LazyCacheRuleSpec;
+import io.quarkus.runtime.ShutdownEvent;
+import io.quarkus.runtime.StartupEvent;
+
+/** Creates two informer on k8s configMap for Lazy and Eager Gingersnap rules
+ * ConfigMap names come from properties
+ * gingersnap.k8s.eager-config-map
+ * gingersnap.k8s.lazy-config-map
+*/
+@ApplicationScoped
+public class CacheRuleInformer {
+   @Inject
+   Instance<KubernetesClient> client;
+   @Inject
+   KubernetesConfiguration configuration;
+   @Inject
+   Event<RuleEvents.LazyRuleAdded> lazyEventAdd;
+   @Inject
+   Event<RuleEvents.LazyRuleRemoved> lazyEventRemove;
+   @Inject
+   Event<RuleEvents.EagerRuleAdded> eagerEventAdd;
+   @Inject
+   Event<RuleEvents.EagerRuleRemoved> eagerEventRemove;
+
+   private SharedIndexInformer<ConfigMap> lazyInformer;
+   private SharedIndexInformer<ConfigMap> eagerInformer;
+
+   private static final Logger log = LoggerFactory.getLogger(CacheRuleInformer.class);
+
+   
+   void startWatching(@Observes StartupEvent ignore) {
+      log.info("startWatching(): begin");
+      if (client.isUnsatisfied()) {
+         log.info("Kubernetes client not found, not watching config map");
+         return;
+      }
+      try {
+         var cmName = configuration.lazyConfigMapName();
+         if (cmName.isEmpty()) {
+            throw new RuntimeException("Lazy configMap not specified, not watching config map");
```

Comment:
We're using Exceptions for flow control here, instead if the ConfigMap name is empty I think we can just have a DEBUG log and continue. That way we can specify only one ConfigMap type during dev.